internal/http/services/owncloud/ocdav/publicfile.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package ocdav
+
+import (
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/rhttp/router"
+	"go.opencensus.io/trace"
+	"net/http"
+	"path"
+)
+
+// PublicFileHandler handles trashbin requests
+type PublicFileHandler struct {
+	namespace string
+}
+
+const (
+	tokenStatInfoKey int = iota
+)
```

Comment:
All this block can be turned into:

```go
type tokenStatInfoKey struct{}
```

and its usage:

```go
tokenStatInfo := ctx.Value(tokenStatInfoKey{}).(*provider.ResourceInfo)
```

Since structures are comparable, we move away from global, since packages can declare their own types.